Presentation.Web/app/shared/main-contract/main-contract-section.component.ts

```diff
@@ -0,0 +1,102 @@
+﻿module Kitos.Shared.Components {
+    "use strict";
+
+    function setupComponent(): ng.IComponentOptions {
+        return {
+            bindings: {
+                pageName: "@",
+                hasWriteAccess: "=",
+                isVisible: "=",
+                viewModel: "<",
+            },
+            controller: MainContractSectionController,
+            controllerAs: "ctrl",
+            templateUrl: `app/shared/main-contract/main-contract-section.view.html`
+        };
+    }
+
+    export interface IMainContractSectionViewModel {
+        isActive: boolean;
+        selectedContractId: number;
+        options: Models.ViewModel.Generic.Select2OptionViewModel<null>[];
+        postMethod: (contractId: number) => ng.IPromise<void>;
+        deleteMethod: () => ng.IPromise<void>;
+        stateReloadMethod: () => ng.IPromise<void>;
+    }
+    
+    interface IMainContractSectionController extends ng.IComponentController {
+        pageName: string;
+        hasWriteAccess: boolean;
+        isVisible: boolean;
+        viewModel: IMainContractSectionViewModel;
+    }
+
+    class MainContractSectionController implements IMainContractSectionController {
+        pageName: string | null = null;
+        hasWriteAccess: boolean | null = null;
+        isVisible: boolean | null = null;
+        viewModel: IMainContractSectionViewModel | null = null;
+        contractModel: {id: number, text: string};
+        currentContractId: number = null;
+        select2Model: Models.ViewModel.Generic.ISingleSelectionWithFixedOptionsViewModel<Models.Generic.NamedEntity.NamedEntityDTO>;
+
+        static $inject: string[] = ["select2LoadingService", "organizationApiService", "notify"];
+        constructor(private readonly select2LoadingService: Services.ISelect2LoadingService) {}
+
+        $onInit() {
+            if (this.pageName === null) {
+                console.error("Missing pageName parameter for MainContractSectionController");
+                return;
+            }
+            if (this.hasWriteAccess === null) {
+                console.error("Missing hasWriteAccess parameter for MainContractSectionController");
+                return;
+            }
+            if (this.isVisible === null) {
+                console.error("Missing isVisible parameter for MainContractSectionController");
+                return;
+            }
+            if (this.viewModel === null) {
+                console.error("Missing viewModel parameter for MainContractSectionController");
+                return;
+            }
+
+            //TODO: is it possible to use "find"? Right now compiler shows an error
+            const selectedContract = this.viewModel.options.filter(x => x.id === this.viewModel.selectedContractId);
```

Comment:
I wanted to use the "find" method (like in the line 61 in the it-system-usage-tab-contracts.controller.ts), but there's a "not found" error here, I guess it's because in the other case the type is "any"

Reply:
@a-naskret It's the TS version... if it's any, the TS compiler pretty much ignores the "error" (when compiled against the ES version we target) - it does not prevent it from working in the browser though.. I think lodash has a helper for it (the `_` framework)